### PR TITLE
Dns names for pods

### PR DESCRIPTION
**Pull request by JonathanO**

### CODE REVIEW

File: controllers/base_controller.go [hunk 1]
```diff
@@ -51,6 +52,10 @@ type ObjectReconciler interface {
 	// Parse the selectors to extract a namespaced name.
 	// For example, a list containing a `k8s_psat:node-name:foo` selector might result in a NamespacedName of "foo"
 	selectorsToNamespacedName([]*common.Selector) *types.NamespacedName
+	// Fill additional fields on a spire registration entry for a k8s object
+	fillEntryForObject(context.Context, *common.RegistrationEntry, ObjectWithMetadata) (*common.RegistrationEntry, error)
+	// Do any additional manager setup required
```
Comment (robinp-tw) [inline controllers/base_controller.go]: nit: Do -> Perform would less likely trip reader into assuming this is a question.

File: controllers/base_controller.go [hunk 2]
```diff
@@ -158,27 +167,43 @@ func (r *BaseReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
...
 func (r *BaseReconciler) entryEquals(myEntry *common.RegistrationEntry, b *common.RegistrationEntry) bool {
 	// TODO: Maybe this should be stricter on the other fields, but right now if you're editing entries on the server, I'll accept that odd things happen to you.
-	return b.SpiffeId == myEntry.GetSpiffeId()
+	if b.SpiffeId != myEntry.GetSpiffeId() {
+		return false
+	}
+	if len(b.DnsNames) != len(myEntry.DnsNames) {
+		return false
+	}
+	stuff := make(map[string]bool, len(b.DnsNames))
```
Comment (robinp-tw) [inline controllers/base_controller.go]: unstuff pls

File: controllers/pod_controller.go [hunk 3]
```diff
@@ -94,6 +104,71 @@ func (r *PodReconciler) makeSpiffeId(obj ObjectWithMetadata) string {
 	return r.makeSpiffeIdForPod(obj.(*corev1.Pod))
 }
 
+func (r *PodReconciler) mungeIp(ip string) string {
+	if strings.Contains(ip, ".") {
+		return strings.Replace(ip, ".", "-", -1)
```
Comment (robinp-tw) [inline controllers/pod_controller.go:110]: No need to check ':' below in this case?

File: controllers/pod_controller.go [hunk 4]
```diff
@@ -165,7 +240,59 @@ func (r *PodReconciler) getObject() ObjectWithMetadata {
 	return &corev1.Pod{}
 }
 
-func NewPodReconciler(client client.Client, log logr.Logger, scheme *runtime.Scheme, trustDomain string, rootId string, spireClient registration.RegistrationClient, mode PodReconcilerMode, value string) *BaseReconciler {
+func (r *PodReconciler) forEachPodSubsetEndpointAddress(subset corev1.EndpointSubset, thing func(corev1.EndpointAddress)) {
```
Comment (robinp-tw) [inline controllers/pod_controller.go]: `thing` -> `f` or `action` or ...?

**Review by robinp-tw — DISMISSED** [review]
Syntactic check mostly.

File: controllers/pod_controller.go (hunk 3 above)
Comment (JonathanO) [reply controllers/pod_controller.go:110]: ip is either going to be IPv4 (contains .) or IPv6 (contains : )

**Review by ttqin — APPROVED** [review]

**Pull request merged**